lib/rules/string.js

```diff
@@ -0,0 +1,392 @@
+const hostnameRegEx = /^(([a-z\d]|[a-z\d][a-z\d\-]*[a-z\d])\.)*([a-z\d]|[a-z\d][a-z\d\-]*[a-z\d])$/i;
+/* eslint-enable max-len */
+
+class StringValidator extends SyncRule {
+    validateSync(params) {
+        return typeof params.value === 'string' && !!params.value;
+    }
+
+    static ruleName() {
+        return 'string';
+    }
+}
+
+/* eslint-disable no-unreachable */
+class Insensitive extends FlagRule {
+    compile(params) {
+        throw new Error('Not implemented');
+        this.invokeLast(params, StringValidator, r => {
+            r.validate = function validate(params2, callback) {
+                const value = params2.toLowerCase();
+                if (this._values.some(v => v.toLowerCase() === value)) {
+                    callback();
+                }
+
+                return callback(this.error(params2));
+            };
+        });
+    }
+
+    static ruleName() {
+        return 'string.insensitive';
+    }
+}
+/* eslint-enable ignore-unreachable */
+
+
+class Min extends SyncRule {
+    compile(params) {
+        this._val = params.args[0];
+    }
+
+    validateSync(params) {
+        return params.value.length >= this._val || {
+            length: params.value.length,
+            min: this._val,
+        };
+    }
+
+    static ruleName() {
+        return 'string.min';
+    }
+}
+
+class Max extends SyncRule {
+    compile(params) {
+        this._val = params.args[0];
+    }
+
+    validateSync(params) {
+        return params.value.length <= this._val || {
+            length: params.value.length,
+            max: this._val,
+        };
+    }
+
+    static ruleName() {
+        return 'string.max';
+    }
+}
+
+class Length extends SyncRule {
+    compile(params) {
+        this._val = params.args[0];
+    }
+
+    validateSync(params) {
+        return params.value.length === this._val || {
+            length: params.value.length,
+            expected: this._val,
+        };
+    }
+
+    static ruleName() {
+        return 'string.length';
+    }
+}
+
+class CreditCard extends SyncRule {
+    validateSync(params) {
+        const str = params.value;
+        if (!ccRegex.test(str)) {
+            return false;
+        }
+        let sum = 0;
+        for (let i = 0; i < str.length; i++) {
+            let intVal = parseInt(str.substr(i, 1), 10);
+            if (i % 2 === 0) {
+                intVal *= 2;
+                if (intVal > 9) {
+                    intVal = 1 + (intVal % 10);
+                }
+            }
+            sum += intVal;
+        }
+        return (sum % 10) === 0;
+    }
+
+    static ruleName() {
+        return 'string.creditCard';
+    }
+}
+
+class RegEx extends SyncRule {
+    compile(params) {
+        this._regEx = params.args[0];
+        this._regExName = params.args[1];
+    }
+
+    validateSync(params) {
+        // TODO: RegEx naming
+        return this._regEx.test(params.value);
```

Comment:
Would probably better to throw an error here because you shouln't be using /g for this kind of validation anyways.